avm/src/vm/deploy.rs

```diff
@@ -0,0 +1,185 @@
+use hyper::{client::Client, Body, Request};
+use serde::{Deserialize, Serialize};
+use serde_json::{from_reader, from_str, json, Value};
+
+use std::error::Error;
+use std::fmt::Display;
+use std::fs::{File, read};
+use std::io::BufReader;
+use std::path::Path;
+
+use ascii_table::{AsciiTable, Column, Align};
+use base64;
+use tokio::runtime::Runtime;
+
+const URL: &str = "http://deploy.alantechnologies.com";
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+struct AWSCredentials {
+  accessKeyId: String,
+  secretAccessKey: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+struct AWSConfig {
+  credentials: AWSCredentials,
+  region: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+struct Config {
+  aws: AWSConfig,
+}
+
+#[derive(Deserialize, Debug)]
+struct App {
+  id: String,
+  url: String,
+  status: String,
+}
+
+const CONFIG_NAME: &str = ".alan/deploy.json";
+const CONFIG_SCHEMA: &str = "Please define one with the following schema: \n{
+  \"aws\": {
+      \"region\": \"string\",
+      \"credentials\": {
+        \"accessKeyId\": \"string\",
+        \"secretAccessKey\": \"string\",
+      }
+  }
+}";
+const HOW_TO_AWS: &str = "
+To create an AWS access key follow this tutorial:\n\nhttps://aws.amazon.com/premiumsupport/knowledge-center/create-access-key/\n\n
+Then enable programmatic access for the IAM user, and attach the built-in 'AdministratorAccess' policy to your IAM user.
+";
+
+fn get_config() -> Config {
+  let home = std::env::var("HOME").unwrap();
+  let file_name = &format!("{}/{}", home, CONFIG_NAME);
+  let path = Path::new(file_name);
+  let file = File::open(path);
+  if file.is_err() {
+    println!("No deploy config at {}", file_name);
```

Comment:
Technically, the config could still be there, but we don't have permissions or there could be some other reason the file failed to open. Maybe print the `Error` instead?

Reply:
Good point